ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,577 @@
+This is a problem because PRs should be assessed while they're being developed and reviewed, active in the teams' minds, instead of some days/weeks later when the team has "paged out" the details of the PR.
+
+It is unlikely that the version control history, the list of pull requests, or any other _automatic_ log can be used to trivialize the obligation of the post hoc assessment.
+Even if we assume an ideal commit messages and/or PR descriptions, there will likely be a burdensome number of them.
+The slightly more realistic assumptions of commit messages and/or PR descriptions that do have ideal content but don't have perfectly uniform structure means the assessor would still have to thoroughly read each, judging what the content requires of the version bump under consideration.
+
+However, there is one standard log that can often make this assessment much easier: the changelog.
+It's not automatic, but it is often structured around "additions/changes/etc", which are exactly what determines the next version under a policy like SemVer, for example.
+
+Thus our final scheme to consider is a variant of FallingEdge that also requires/assumes the maintenance of a _bumplog_, which is merely a degenerate changelog that contains nothing more than what's necessary to determine the next version number for each changed packaged (ie one classification of `patch`, `minor`, and `major` for each merged PR).
+
+| TypicalReleases | EasyPR | EasyRelease |
+|:---------------:|:------:|:-----------:|
+| ✓               | 90% ✓  | 95% ✓       |
+
+We assign ✓ for TypicalRelease for the same reason as FallingEdge.
+We assign a partial 90% ✓ for EasyPR because the presence of a correct bumplog entry requires non-trivial effort from the submitter and the reviewer.
+We also assign a partial 95% ✓ for EasyRelease to account for the following small checklist for each release.
+- Scan the bumplog entries that are new since the previous release from this branch.
+  (We the degenerately minimal structure of the entries ensures this is easy.)
+- Thereby determine what the next version should be for each package being released.
+- Add a commit that sets those versions accordingly.
+
+If you're envisioning a single bumplog file, then you might be wondering why this scheme wouldn't also incur spurious merge conflicts among PRs, etc.
+Fortunately, this is already a solved problem for changelogs; see tools like https://scriv.readthedocs.io/en/latest et al.
+
+_Remark_.
+We chose 90% and 95% based on the use of a degenerate changelog.
+Mature/stable projects likely already have a non-degenerate changelog that would make the pre-release assessment required by FallingEdge very simple, if not totally trivial.
+Thus, such projects' teams might assign slighty better and/or worse scores to EasyPR and/or
+EasyRelease to this scheme, depending on how they choose to resolve the tension between the bumplog and the structure/thoroughness of their actual changelog.
```

Comment:
Perhaps you might consider it a point in favour of having the changelog and bumplog merged that it nudges people to make changelog entries! It's easy to think "oh, this doesn't need a changelog entry, it's so minor", but if you have to edit it _anyway_ to say "patch bump"... you might as well write a sentence about what changed.

Reply:
I've been thinking about exactly the same, e.g. the next unpublished section of changelog could be split into two lists of major and minor changes.